rest/route/project.go

```diff
@@ -508,9 +508,12 @@ func (h *projectIDPatchHandler) Run(ctx context.Context) gimlet.Responder {
 		return gimlet.MakeJSONErrorResponder(errors.Wrap(catcher.Resolve(), "error validating triggers"))
 	}
 
-	err = dbModel.BbProjectIsValid(h.newProjectRef.Id, h.newProjectRef.BuildBaronSettings)
-	if err != nil {
-		return gimlet.MakeJSONErrorResponder(errors.Wrap(err, "error validating build baron config"))
+	errs := dbModel.ValidateBbProject(h.newProjectRef.Id, h.newProjectRef.BuildBaronSettings, nil, true)
```

Comment:
Should this be using newProjectRef.TaskAnnotationSettings.Webhook instead of looking it up?

Reply:
Yes since we actually do have access to the webhook in all places where we call ValidateBbProject, we can directly pass it in for all cases.  Since FindMergedProjectRef already prevents webhook & BB config from picking up the project config values if webhook & BB are configured for the project ref we don't need to consider that case during validation.  Changing to pass in newProjectRef.TaskAnnotationSettings.Webhook